cuda_core/docs/source/interoperability.rst

```diff
@@ -52,7 +52,8 @@ Then such objects can be understood by ``cuda.core`` anywhere a stream-like obje
 is needed.
```

Comment:
I think we should discuss this. This will introduce non-negligible overhead, similar to DLPack, where we might be better off having a user explicitly wrap their stream objects in a `cuda.core.Stream` and reuse that `Stream` object as needed.

Reply:
Yes right now only `launch()` and `Device.create_stream()` support the protocol, due to the exact concern of adding overheads. Overhead aside, I think essentially we are assessing which interoperability story we want to tell: 
- We give people a convenient way for them to convert their types to ours, and only accept our types in our APIs
- We accept everyone's types in our APIs

Both stories need the protocol so it's not a waste. Just how to use the protocol is different.

Reply:
I'll work on 1 now.